gabbi/tests/test_runner.py

```diff
@@ -14,31 +14,61 @@
 """
 
 from io import StringIO
+import os
+import socket
+import subprocess
 import sys
+import time
 import unittest
-from uuid import uuid4
-
-from wsgi_intercept.interceptor import Urllib3Interceptor
 
 from gabbi import exception
 from gabbi.handlers import base
 from gabbi.handlers.jsonhandler import JSONHandler
 from gabbi import runner
-from gabbi.tests.simple_wsgi import SimpleWsgi
+
+
+def get_free_port():
+    sock = socket.socket()
+    sock.bind(('', 0))
+    return sock.getsockname()[1]
+
+
+class ForkedWSGIServer:
+
+    def __init__(self, host, port):
+        self.host = host
+        self.port = port
+
+    def start(self):
+        self.process = subprocess.Popen(
+            [
+                "python",
+                "gabbi/tests/external_server.py",
+                self.host,
+                str(self.port)
+            ],
+            env=os.environ.update({"PYTHONPATH": "."}),
+            close_fds=True,
+        )
+        # We need to sleep a bit to let the wsgi server start.
+        # TODO(cdent): This is regrettable.
+        time.sleep(.4)
+
+    def stop(self):
+        self.process.terminate()
 
 
 class RunnerTest(unittest.TestCase):
 
+    port = get_free_port()
+
     def setUp(self):
         super(RunnerTest, self).setUp()
 
-        # NB: random host ensures that we're not accidentally connecting to an
-        #     actual server
-        host, port = (str(uuid4()), 8000)
-        self.host = host
-        self.port = port
-        self.server = lambda: Urllib3Interceptor(
-            SimpleWsgi, host=host, port=port)
+        self.host = "0.0.0.0"
```

Comment:
Just to be safe: Do we want 0.0.0.0 or localhost here?

More generally, I'm not sure why this needed changing in this patch.

Reply:
This ended up 0.0.0.0 as a result of the dancing done to get full url resolution working the same on my Mac versus the CI nodes. I'll make some comments about that.